server/balancer.go

```diff
@@ -370,3 +372,254 @@ func (r *replicaChecker) checkBestReplacement(region *RegionInfo) Operator {
+	antiCount      int
+	version        uint64
+}
+
+// ListRegionsStat is a list of a group region state type
+type ListRegionsStat []RegionStat
+
+func (m ListRegionsStat) Len() int           { return len(m) }
+func (m ListRegionsStat) Swap(i, j int)      { m[i], m[j] = m[j], m[i] }
+func (m ListRegionsStat) Less(i, j int) bool { return m[i].WrittenBytes < m[j].WrittenBytes }
+
+// StoreHotRegions records all hot regions in one store with sequence
+type StoreHotRegions struct {
+	StoreTotalWrittenBytes uint64          `json:"total_written"`
+	RegionsNumber          int             `json:"regions_number"`
+	RegionsStat            ListRegionsStat `json:"status"`
+}
+
+type balanceHotRegionScheduler struct {
+	opt         *scheduleOption
+	limit       uint64
+	scoreStatus map[uint64]*StoreHotRegions // store id -> regions status in this store
+	r           *rand.Rand
+}
+
+func newBalanceHotRegionScheduler(opt *scheduleOption) *balanceHotRegionScheduler {
+	return &balanceHotRegionScheduler{
+		opt:         opt,
+		limit:       1,
+		scoreStatus: make(map[uint64]*StoreHotRegions),
+		r:           rand.New(rand.NewSource(time.Now().UnixNano())),
+	}
+}
+
+func (l *balanceHotRegionScheduler) GetName() string {
+	return "balance-hot-region-scheduler"
+}
+
+func (l *balanceHotRegionScheduler) GetResourceKind() ResourceKind {
+	return priorityKind
+}
+
+func (l *balanceHotRegionScheduler) GetResourceLimit() uint64 {
+	return l.limit
+}
+
+func (l *balanceHotRegionScheduler) Prepare(cluster *clusterInfo) error { return nil }
+
+func (l *balanceHotRegionScheduler) Cleanup(cluster *clusterInfo) {}
+
+func (l *balanceHotRegionScheduler) Schedule(cluster *clusterInfo) Operator {
+	l.CalculateScore(cluster)
+	region := l.SelectSourceRegion(cluster)
+	if region == nil {
+		return nil
+	}
+	newLeader := l.selectTransferLeader(region)
+	if newLeader != nil {
+		return newPriorityTransferLeader(region, newLeader)
+	}
+	peer := l.selectTransferPeer(region, cluster)
+	if peer != nil {
+		return newPriorityTransferPeer(region, region.Leader, peer)
+	}
+	return nil
+}
+
+func (l *balanceHotRegionScheduler) clearScore() {
+	l.scoreStatus = make(map[uint64]*StoreHotRegions)
+}
+func (l *balanceHotRegionScheduler) CalculateScore(cluster *clusterInfo) {
+	l.clearScore()
+	items := cluster.writeStatistics.elems()
+	for _, item := range items {
+		r, ok := item.value.(RegionStat)
+		if !ok {
+			continue
+		}
+		if r.HotDegree < hotRegionLowThreshold {
+			continue
+		}
+
+		regionInfo := cluster.getRegion(r.RegionID)
+		storeID := regionInfo.Leader.GetStoreId()
+		status, ok := l.scoreStatus[storeID]
+		if !ok {
+			status = &StoreHotRegions{
+				RegionsStat: make(ListRegionsStat, 0, 100),
+			}
+			l.scoreStatus[storeID] = status
+		}
+		status.StoreTotalWrittenBytes += r.WrittenBytes
+		status.RegionsStat = append(status.RegionsStat, RegionStat{r.RegionID, r.WrittenBytes, r.HotDegree, r.LastUpdateTime, storeID, r.antiCount, r.version})
```

Comment:
please use field name initialization to avoid filling the fields with incorrect order. 

like 
```
RegionStat {
    RegionID: r.RegionID,
    ...
}
```